app/grandchallenge/notifications/models.py

```diff
@@ -37,3 +129,274 @@ def _assign_permissions(self):
         assign_perm("view_notification", self.user, self)
         assign_perm("delete_notification", self.user, self)
         assign_perm("change_notification", self.user, self)
+
+    @staticmethod
+    def send(type, **kwargs):  # noqa: C901
+        verb = kwargs.pop("verb")
+        try:
+            actor = kwargs.pop("actor")
```

Comment:
You can get rid of all these `try.. except` statements by providing a default for `pop`, e.g. `actor = kwargs.pop("actor", None)`.

Reply:
ah that's a lot better, thanks!